source/back.tex

```diff
@@ -158,6 +158,15 @@
 \printindex[conceptindex]
 }
 
+\clearpage
+\renewcommand{\preindexhook}{The bold page number for each entry is the page
+where the macro is defined.
+Other page numbers refer to pages where the macro is mentioned in the general text.}
```

Comment:
The text describing the index is not accurate, as the LaTeX macros do not support annotating a primary definition.  It is not clear that the primary definition language will be useful for C++ macros either.

I failed to come up with a better introductory paragraph myself though, as all of the other indexes start with specific text that does not apply here either.

Reply:
> The text describing the index is not accurate, as the LaTeX macros do not support annotating a primary definition.

Well, for the library concepts index, we do have a bold page number for the primary definition. (I haven't looked at the actual LaTeX macros proposed here.)

Reply:
Added `\defmacroname` for where the macro semantics is defined
(might be in prose, not necessarily a `#define`).